playbook/app/pb_kits/playbook/pb_phone_number_input/_phone_number_input.tsx

```diff
@@ -0,0 +1,155 @@
+  }
+}
+
+type PhoneNumberInputProps = {
+  aria?: { [key: string]: string },
+  className?: string,
+  data?: { [key: string]: string },
+  disabled?: boolean,
+  id?: string,
+  initialCountry?: string,
+  isValid?: (valid: boolean) => void,
+  label?: string,
+  name?: string,
+  onChange?: (e: React.FormEvent<HTMLInputElement>) => void,
+  onlyCountries: string[],
+  preferredCountries?: string[],
+  value?: string,
+}
+
+enum ValidationError {
+  TooShort = 2,
+  TooLong = 3,
+}
+
+const formatToGlobalCountryName = (countryName: string) => {
+  return countryName.split('(')[0].trim()
+}
+
+const formatAllCountries = () => {
+  let countryData = window.intlTelInputGlobals.getCountryData()
+  for (let i = 0; i < countryData.length; i++) {
+    let country = countryData[i]
+    country.name = formatToGlobalCountryName(country.name)
+  }
+}
+
+const containOnlyNumbers = (value: string) => {
+  return /^(\++)*(\d+)$/.test(value)
+}
+
+const PhoneNumberInput = (props: PhoneNumberInputProps) => {
+  const {
+    aria = {},
+    className,
+    data = {},
+    disabled = false,
+    id = '',
+    initialCountry = '',
+    isValid = () => {void 0 },
+    label = '',
+    name = '',
+    onChange = () => { void 0 },
+    onlyCountries = [],
+    preferredCountries = [],
+    value = '',
+  } = props
+
+  const ariaProps = buildAriaProps(aria)
+  const dataProps = buildDataProps(data)
+  const classes = classnames(buildCss('pb_phone_number_input'), globalProps(props), className)
+
+  const inputRef = useRef<HTMLInputElement>()
+  const [inputValue, setInputValue] = useState(value)
+  const [itiInit, setItiInit] = useState<any>()
+  const [error, setError] = useState('')
+
+  const validateTooLongNumber = (itiInit: any) => {
+    const error = itiInit.getValidationError()
+
+    if (error === ValidationError.TooLong) {
+      const countryName = itiInit.getSelectedCountryData().name
+      setError(`Invalid ${countryName} phone number (too long)`)
+    } else {
+      setError('')
+    }
+  }
+
+  const validateTooShortNumber = () => {
+    const error = itiInit.getValidationError()
+
+    if (error === ValidationError.TooShort) {
+      const countryName = itiInit.getSelectedCountryData().name
+      setError(`Invalid ${countryName} phone number (too short)`)
+    }
+  }
+
+  const validateOnlyNumbers = () => {
+    if (inputValue && !containOnlyNumbers(inputValue)) {
+      setError('Invalid phone number. Enter numbers only.')
+    }
+  }
+
+  const validateErrors = () => {
+    validateTooShortNumber()
+    validateOnlyNumbers()
+  }
+
+  const handleOnChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(evt.target.value)
+    validateTooLongNumber(itiInit)
+    onChange(evt)
+    isValid(itiInit.isValidNumber())
+  }
+
+  useEffect(() => {
+    formatAllCountries()
+
+    const telInputInit = new intlTelInput(inputRef.current, {
+        utilsScript: 'https://cdnjs.cloudflare.com/ajax/libs/intl-tel-input/17.0.19/js/utils.js',
```

Comment:
This seems like a bad idea. Is there not an NPM package?

Reply:
We are using the npm package for the intl-tel-input, [here](https://github.com/powerhome/playbook/blob/b9b05cd3dde5687b84c590fd65f8eca9fc3b211d/playbook/package.json#L47).

For the utils script, there is not a specific npm package.
```
The utilities script ([build/js/utils.js](https://github.com/jackocnr/intl-tel-input/blob/master/build/js/utils.js)) is a custom build of Google's [libphonenumber](https://github.com/googlei18n/libphonenumber)
```

Reply:
Yeah @carloslimasd to Jason's point, did you try pointing this to the `build/js/utils.js` location?
It's in their docs here:
<img width="1141" alt="Screen Shot 2023-02-17 at 2 16 01 PM" src="https://user-images.githubusercontent.com/9158723/219766359-16df3535-5f70-4793-9c99-e44eb4480ee4.png">